tools/hardware/outputs/c_header.py

```diff
@@ -36,6 +36,15 @@
 #include <linker.h>         /* for BOOT_RODATA */
 #include <basic_types.h>    /* for p_region_t, kernel_frame_t (arch/types.h) */
 
+/* Wrap raw physBase location constant to give it a symbolic name in C that's
+ * visible to verification. This is necessary as there are no real constants
+ * in C except enums, and enums constants must fit in an int.
+ */
+static inline CONST word_t physBase(void)
+{
+    return PHYS_BASE_RAW;
+}
+
```

Comment:
It seems all this affects 32-bit platforms only. On 64-bit architectures we define PADDR_BASE to be UL_CONST(0), so this is not used at all. Should we have `#define PADDR_BASE physBase()' here to ensure this is consistent everywhere? Or is this all something to be buried in the 32-bit world?

Reply:
For PADDR_BASE, it does seem to affect 32 bit only, but it affects KERNEL_ELF_PADDR_BASE and with that KERNEL_ELF_BASE and `addrFromKPPtr` on 64 bit platforms as well.

Reply:
physBase comes from defining the base physical address of the kernel virtual mapping. If the kernel virtual mapping domain is greater or equal to the range of physical addresses, then having the base of the virtual domain map to the lowest physical address seems simplest.

When the physical address range is larger than the kernel virtual mapping (which on 32bit it is) then we can only map to a subset of the physical address range. 

If we expect that on all 64bit architectures the physical address range will fit within the virtual mapping space then physBase would only ever need to be 0. But who knows if this will always be the case as sufficiently complex devices start requiring larger address ranges for encoding more information than just a spatial address into the physical address.

Reply:
Ok, got it, that makes perfect sense, yes. I guess we can change it when we get to a device that needs it. For now it seems reasonable to keep 0 here.